### ISSUE TITLE
Create custom entity for metastore

### ISSUE THREAD

**Issue by dafeder** [GetDKAN/dkan#3321]
Build out a custom entity with bundles to replace the "data" content type in Drupal.

## Acceptance criteria

- On enabling module, there is a custom entity for metastore items and bundles for schemas
- Clear admin interface that is at least as intuitive as existing node one.
- All DKAN functions work seamlessly with new entity system.
- I can see readable titles for all my metadata so that I can find what I'm looking for.
- Hashes are storable and retrievable. This will be done in #3465 so this ticket can't be merged until that's done.

## Estimate: 
2 days

**Comment by dafeder** [comment]
I've gotten pretty far with this but the acceptance criteria are blocked by #3444. Putting this aside until that's been done. This may actually be complete and just need the lifecycle and reference work to start passing tests.

Once #3444 is in this should take no more than 1 day to complete.

**Comment by ghost** [comment]
From Dan: Finishing #3321 is too ambitious for the timeline that we have (we’re 75% there). We should deprioritize this one and write up another ticket to do the admin work that is absolutely necessary for this sprint. This new ticket can be done today.

**Comment by dafeder** [comment]
See #3168 for some background
